Edit tctl instructions to clarify remote login

ptgott opened the pull request:
Closes #11464

- Ensure that all example tctl commands are accompanied either by
  instructions to log in to the cluster or the tctl.mdx partial.

- Edit the guides in the Architecture section to remove notes that
tctl can only be used locally to the Auth Service.

- Edit the user-client-rereqs partial to mention tctl for all
  editions, since you can log in to tctl remotely for all editions.

Not editing guides where:

- tctl is run via kubectl exec
- tctl is not mentioned in a code block, i.e., only in passing, and
  a reader isn't expected to run the command on their own while
  following the guide.
- The user is already expected to run tctl on the Auth Service. The
  docker-compose Getting Started guide is an example of this.

ptgott committed:
Edit tctl instructions to clarify remote login

Closes #11464

- Ensure that all example tctl commands are accompanied either by
  instructions to log in to the cluster or the tctl.mdx partial.

- Edit the guides in the Architecture section to remove notes that
tctl can only be used locally to the Auth Service.

- Edit the user-client-rereqs partial to mention tctl for all
  editions, since you can log in to tctl remotely for all editions.

Not editing guides where:

- tctl is run via kubectl exec
- tctl is not mentioned in a code block, i.e., only in passing, and
  a reader isn't expected to run the command on their own while
  following the guide.
- The user is already expected to run tctl on the Auth Service. The
  docker-compose Getting Started guide is an example of this.

zmb3 commented on docs/pages/access-controls/reference.mdx:
````diff
@@ -33,10 +34,26 @@ To manage cluster roles, a Teleport administrator can use the Web UI or the
+<ScopedBlock scope={["oss", "enterprise"]}>
+
 ```code
+# Log in to your cluster so you can use tctl.
````
```suggestion
# Log in to your cluster with tsh so you can use tctl.
```

Let's be extra explicit that the login is with tsh so that people don't mistake this for an error.

zmb3 commented on docs/pages/architecture/overview.mdx:
```diff
@@ -234,11 +227,11 @@ valid for 12 hours by default, unless you specify another interval via the
 You can learn more about `tsh` in the [User Manual](../server-access/guides/tsh.mdx).
 
-### TCTL
+### `tctl`
```
Should we add a paragraph here about authenticating?

I think it would be worth mentioning that:
- tctl can be run on the auth server directly, if the config file is specified.
- tctl can pick up a tsh profile automatically

ptgott committed:
Respond to PR feedback

- Provide more context on authenticating with tctl in the CLI reference
- Update the link to more information re: tctl in the architecture
  overview, and indicate that tctl users must authenticate.
- Minor tweaks.

r0mant commented on docs/pages/application-access/guides/dynamic-registration.mdx:
````diff
@@ -82,10 +82,27 @@ version: v5
+
 ```code
+# Log in to your Teleport cluster so you can use tctl remotely.
+$ tsh login --proxy=mytenant.teleport.sh --user=myuser
````
This is just a meta thought but I wonder if there is a less verbose way for us to modify parts of the commands based on the selected scope. Right now we have to copy entire blocks when the only part that's different it the "proxy" value. E.g. can we have a "proxy" config variable that would change value based on the scope? Not sure if our engine supports that. Would also help with the consistency of examples as we'd use the same "example" proxy everywhere.

ptgott replied:
Our variable replacement code is a [`remark`](https://github.com/remarkjs/remark) plugin that [replaces](https://github.com/gravitational/docs/blob/664b6a324c5d07e52ac1f4ecf76348e8d4c8cf84/server/remark-variables.ts#L47-L51) any instance of `(= varname =)` with the value provided in `config.json`. `remark` runs before the Markdown in our docs is converted into JSX (i.e., before scopes are assigned), so we would need to come up with a new way to populate variables based on scope.

I agree that we do have a lot of these `ScopedBlocks` that do nothing but rename the Proxy address, and it does get tedious to write and review. I'll create a ticket with the web team to look into this!

r0mant commented on docs/pages/architecture/overview.mdx:
```diff
@@ -234,16 +227,17 @@ valid for 12 hours by default, unless you specify another interval via the
 
-`tctl` is also a tool that can be used to modify the dynamic configuration of
-the cluster, like creating new user roles or connecting trusted clusters.
+When running `tctl`, a user must authenticate to a Teleport cluster. `tctl` can
```
I would tweak the wording here a bit. When running on an auth server directly, tctl will use the identity of the auth server itself when talking to its API, you don't need any extra authentication. When running on a client host, user needs to log into the cluster with tsh and tctl will use the user's identity created by tsh login.

ptgott committed:
Respond to PR feedback

- Rephrase the authentication paragraph in the Architecture Overview.
